aten/src/ATen/Context.cpp

```diff
@@ -24,20 +24,14 @@ static std::vector<std::string> generic_precisions = {"default", "tf32", "bf16"}
 static std::vector<std::string> cuda_precisions = {"default", "tf32"};
 static std::vector<std::string> mkldnn_precisions = {"default", "bf16"};
 
-static void warn_invalid_fp32_precision(const std::string& backend, std::string& precision) {
-  bool invalid_cuda = backend == "cuda" && std::find(cuda_precisions.begin(), cuda_precisions.end(), precision) == cuda_precisions.end();
-  bool invalid_mkldnn = backend == "mkldnn" && std::find(mkldnn_precisions.begin(), mkldnn_precisions.end(), precision) == mkldnn_precisions.end();
-  if (invalid_cuda || invalid_mkldnn){
-    precision = "default";
-    TORCH_WARN("Invalid precision for ", backend, " precision: ", precision, " reset precision to default");
-  }
-}
-
-static void fp32_precision_valid_check(
+// This method first check whether the backend and operators are legal
+// After that, if there is non-empty precision
+// it will futher check whether the precision is legal
+// and return the result
+static bool validate_fp32_prec_with_backend_and_op(
```

Comment:
Is it cleaner to separate the "check" with assertions and the check on precision that returns boolean values?